Library/Homebrew/os/mac.rb

```diff
@@ -37,6 +37,15 @@ def outdated_release?
       version < "10.9"
     end
 
+    def os_name(version = @version)
+      # TODO: replace os_name by constant string "macOS" when macOS 10.14 is released
+      if Version.create(version) >= Version.create("10.12")
```

Comment:
@MikeMcQuaid Thank you! `Version.new` LGTM, I'd better add more test to ensure it does not break.